Source/WebKit/WebProcess/GPU/graphics/RemoteImageBufferProxy.h

```diff
@@ -327,13 +334,26 @@ class RemoteImageBufferProxy : public WebCore::ConcreteImageBuffer<BackendType>
         return WTF::makeUnique<ThreadSafeRemoteImageBufferFlusher<BackendType>>(*this);
     }
 
+    void prepareForBackingStoreChange()
+    {
+        if (LIKELY(!m_mayHaveOutstandingBackingStoreReferences))
+            return;
+        ASSERT(!hasPendingFlush());
+        auto* backend = ensureBackendCreated();
+        if (UNLIKELY(!backend))
+            return;
+        backend->ensureNativeImagesHaveCopiedBackingStore();
+        m_mayHaveOutstandingBackingStoreReferences = false;
+    }
+
     WebCore::GraphicsContextFlushIdentifier m_sentFlushIdentifier;
     Lock m_receivedFlushIdentifierLock;
     Condition m_receivedFlushIdentifierChangedCondition;
     WebCore::GraphicsContextFlushIdentifier m_receivedFlushIdentifier WTF_GUARDED_BY_LOCK(m_receivedFlushIdentifierLock); // Only modified on the main thread but may get queried on a secondary thread.
     WeakPtr<RemoteRenderingBackendProxy> m_remoteRenderingBackendProxy;
     RemoteDisplayListRecorderProxy m_remoteDisplayList;
     bool m_needsFlush { false };
+    mutable bool m_mayHaveOutstandingBackingStoreReferences { false };
```

Comment:
Should this flag be moved to `ImageBufferIOSurfaceBackend` since it seems a backend property and it is related to IOSurface only? It is set to true in `RemoteImageBufferProxy::copyNativeImage()` after calling `ImageBufferIOSurfaceBackend::copyNativeImage()` via `ConcreteImageBuffer::copyNativeImage()`. And it is reset to false after calling `ImageBufferIOSurfaceBackend::ensureNativeImagesHaveCopiedBackingStore()` in `prepareForBackingStoreChange()`.

Reply:
Good idea, done.